src/render/systems.rs

```diff
@@ -241,42 +241,65 @@ pub fn render_frame(
         .unwrap();
 }
 
+// Returns the width and heigh of the available viewport space;
+// camera viewport size if present, otherwise default to window size
+pub fn get_viewport_size(
+    camera_query: Query<&Camera, With<VelloView>>,
+    window: Option<Single<&Window, With<PrimaryWindow>>>,
+) -> (u32, u32) {
+    if let Ok(camera) = camera_query.single() {
+        if let Some(viewport) = &camera.viewport {
+            return (viewport.physical_size.x, viewport.physical_size.y);
+        }
+    }
+
+    let Some(window) = window.as_deref() else {
+        panic!("We only support rendering to the primary window right now.");
+    };
+    (
+        window.resolution.physical_width(),
+        window.resolution.physical_height(),
+    )
+}
+
 pub fn resize_rendertargets(
-    mut window_resize_events: EventReader<WindowResized>,
+    window_resize_events: EventReader<WindowResized>,
```

Comment:
Is there a better event, to determine if the viewport changes size? Or we can make our own event for this perhaps?